engine/src/main/java/org/terasology/rendering/dag/RenderTaskListGenerator.java

```diff
@@ -15,41 +15,49 @@
  */
 package org.terasology.rendering.dag;
 
+import com.google.common.collect.Lists;
 import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * TODO: Add javadocs
  */
 public final class RenderTaskListGenerator {
+    private static final Logger logger = LoggerFactory.getLogger(RenderTaskListGenerator.class);
 
     public RenderTaskListGenerator() {
 
     }
 
     public List<RenderPipelineTask> generateFrom(List<Node> orderedNodes) {
-        List<RenderPipelineTask> taskList = com.google.common.collect.Lists.newArrayList();
+        logger.info("Task List: "); // TODO: remove in the future or turn it into debug
+        List<RenderPipelineTask> taskList = Lists.newArrayList();
         for (Node node : orderedNodes) {
-            // TODO: add "renderPipeline.addAll(node.getDesiredStateChanges())" here
+
```

Comment:
We are missing a step here!

From orderedNodes we must first generate a List<Object> containing StateChanges and Nodes. This list also includes the state changes to return parameters back to defaults. 

Once that list is in place we can generate the tasklist, making sure we do not include redundant tasks.